test/compilable/testclidflags.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+dir=${RESULTS_DIR}${SEP}compilable
+output_file=${dir}/testclidflags.sh.out
+
+unset DFLAGS
+
+# Force DMD to print the -v menu by passing an invalid object file
+# It will fail with "no object files to link", but print the log
+# On OSX DMD terminates with a successful exit code, so `|| true` is used.
+( "$DMD" -v foo.d 2> /dev/null || true) | grep -q "DFLAGS    (none)"
+( DFLAGS="-O -D" "$DMD" -v foo.d 2> /dev/null || true) | grep -q "DFLAGS    -O -D"
+( DFLAGS="-O '-Ifoo bar' -c" "$DMD" -v foo.d 2> /dev/null || true) | grep -q "DFLAGS    -O '-Ifoo bar' -c"
```

Comment:
I wonder why this works on the auto tester. For me, this test fails without any helpful output and if I run it explicitly it seems DFLAGS is picked up as a test parameter and d_do_test complains:
```
object.Exception@d_do_test.d(584): The DFLAGS test argument must be empty: It is '   ="-O -D" "$DMD" -v foo.d 2> /dev/null || true) | grep -q "DFLAGS    -O -D" ="-O '-Ifoo bar' -c" "$DMD" -v foo.d 2> /dev/null || true) | grep -q "DFLAGS    -O '-Ifoo bar' -c"'
```

Reply:
You running Windows?

Reply:
Yes, but picking up DFLAGS should be OS platfrom-agnostic.

Reply:
Maybe something to do with the binutils or bash? What's the actual output of `dmd` for you?

Reply:
It's not dmd complaining, but d_do_test because `findTestParameter(envData, file, "DFLAGS", testArgs.dflags);` results in the shown string.

Reply:
Sorry, that's weird. Bash parameters don't outlive their scope/script.
Does `unset DFLAGS` at the end help?

Reply:
It doesn't have to do with BASH.

`d_do_test` parses test parameters from the file and it looks like it is "accidently" picking up the new `DFLAGS` parameter from the BASH script itself...which it shouldn't be doing.

Reply:
Oh wait...`d_do_test` isn't involved in BASH tests at all...

@rainers are you sure it's this test that is failing?  If so, then the Makefile is probably using the wrong rule to run this test.

Reply:
Ouch, totally missed that shell scripts are (no longer?) executed through d_do_test. 
There is zero output from the script itself, will have to investigate further...

Reply:
We just merged a chnage to BASH scripts which adds `sh_do_test.sh` that automatically redirects output from all BASH tests to the test output file, i.e. `test_reuslts/compilable/mytest.sh.out`

Reply:
https://github.com/dlang/dmd/pull/7928

Reply:
Found the issue: dmd still used an sc.ini which overrides DFLAGS. Better to pass `-conf=` to dmd in these tests, too?

Reply:
https://github.com/dlang/dmd/pull/7966